go/vt/mysqlctl/s3backupstorage/s3.go

```diff
@@ -80,17 +81,30 @@ func (bh *S3BackupHandle) Name() string {
 }
 
 // AddFile is part of the backupstorage.BackupHandle interface.
-func (bh *S3BackupHandle) AddFile(ctx context.Context, filename string) (io.WriteCloser, error) {
+func (bh *S3BackupHandle) AddFile(ctx context.Context, filename string, filesize int64) (io.WriteCloser, error) {
 	if bh.readOnly {
 		return nil, fmt.Errorf("AddFile cannot be called on read-only backup")
 	}
 
+	// Calculate s3 upload part size using the source filesize
+	partSizeMB := s3manager.DefaultUploadPartSize
+	if filesize > 0 {
+		minimumPartSize := float64(filesize) / float64(s3manager.MaxUploadParts)
```

Comment:
I'm not sure about dynamically changing the part size without specifying a max limit.  The problem with tuning the part size is now you have to be able to fit that in memory -- multiplied by the concurrency.  So I think there should be a flag for the max part size and a flag for max upload parts. Using those 2 along with concurrency, someone can tune the memory footprint of their backups

Reply:
Given a 120gb table, that would be 12mb parts. For that to have any kind of significant impact you would have to set concurrency pretty high.

I didn't think this change would affect anyone, given that tables under 50gb would just be using the same 5mb default part size as before and tables over 50gb currently don't even upload.

Reply:
Just to add, I was also cautious of making a blanket part size flag because then that would create large upload buffers for files that are very small. I really wanted to use a larger part size only for files that needed it and leave the rest to the default size of 5mb.

Reply:
looks like MaxUploadParts is not changeable, so this is probably the only course of action.  🤞